File: analysis/import/SCMLogParser/src/test/java/de/maibornwolff/codecharta/importer/scmlogparser/parser/git/CommitDateParserTest.java
package de.maibornwolff.codecharta.importer.scmlogparser.parser.git;

import org.junit.Test;

import java.time.LocalDateTime;

import static org.assertj.core.api.Assertions.assertThat;

public class CommitDateParserTest {
    @Test
    public void parseCommitDate() {
        LocalDateTime date = CommitDateParser.parseCommitDate("Date:   Tue May 9 19:57:57 2017 +0200");
        assertThat(date).isEqualTo(LocalDateTime.of(2017, 5, 9, 19, 57, 57));
    }
}

File: analysis/import/SCMLogParser/src/test/kotlin/de/maibornwolff/codecharta/importer/scmlogparser/parser/git/CommitDateParserTest.kt
package de.maibornwolff.codecharta.importer.scmlogparser.parser.git

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import java.time.OffsetDateTime
import java.time.ZoneOffset

class CommitDateParserTest {
    @Test
    fun parseCommitDate() {
        val date = CommitDateParser.parseCommitDate("Date:   Tue May 9 19:57:57 2017 +0200")
        assertThat(date).isEqualToIgnoringNanos(OffsetDateTime.of(2017, 5, 9, 19, 57, 57, 0, ZoneOffset.ofHours(2)))
    }
}